packer/scripts/macos/macos-agentsetup.sh

```diff
@@ -13,28 +13,41 @@ sudo chown -R ec2-user:staff /var/jenkins
 /usr/local/bin/brew install grep
 /usr/local/bin/brew install wget 
 /usr/local/bin/brew install maven
+/usr/local/bin/brew install dpkg
 
-## Install MacPorts, jEnv, setup java8,11,17,21 and python, then set default java to 11
+## Array of JDK versions in form of version@URL@priority
+jdk_versions=(
+    "8@https://github.com/adoptium/temurin8-binaries/releases/download/jdk8u392-b08/OpenJDK8U-jdk_x64_mac_hotspot_8u392b08.tar.gz@1"
+    "11@https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.21%2B9/OpenJDK11U-jdk_x64_mac_hotspot_11.0.21_9.tar.gz@100"
+    "17@https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.9%2B9/OpenJDK17U-jdk_x64_mac_hotspot_17.0.9_9.tar.gz@1"
+    "19@https://github.com/adoptium/temurin19-binaries/releases/download/jdk-19.0.2%2B7/OpenJDK19U-jdk_x64_mac_hotspot_19.0.2_7.tar.gz@1"
+    "20@https://github.com/adoptium/temurin20-binaries/releases/download/jdk-20.0.2%2B9/OpenJDK20U-jdk_x64_mac_hotspot_20.0.2_9.tar.gz@1"
+    "21@https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1%2B12/OpenJDK21U-jdk_x64_mac_hotspot_21.0.1_12.tar.gz@1"
+)
+
+## Loop through JDK versions and install them
+for version_info in "${jdk_versions[@]}"; do
+    IFS='@' read -ra version_array <<< "$version_info"
+    version_num="${version_array[0]}"
+    version_url="${version_array[1]}"
+    version_priority="${version_array[2]}"
```

Comment:
Suggest using cut or awk, not sure why it is using IFS now.

Reply:
oh,, I did change it to be using IFS as I thought it would be more efficient doing this way than using cut ,, but yes , I can certainly switch it back to use cut :)